### PR TITLE
followup to incorporate review comments per previous reviews on PR#29561

### PR DESCRIPTION
**Pull request by londoncalling**
### What I did

Incorporated review comments

### Description for the changelog

Glossary updates for Docker for Mac and Windows, copy-on-write, and union filesystem

### CODE REVIEW

**Comment by londoncalling** [comment]
@thaJeztah @mstanleyjones these are the changes that didn't make it in before PR #29561 got merged, but now I am getting conflicts that don't show up when I fetch and rebase. @mstanleyjones let's look at this together tomorrow.

**Comment by allencloud** [comment]
Conflict occurs, rebase need, I am afraid @londoncalling  😄

**Comment by londoncalling** [comment]
@thaJeztah @mstanleyjones @allencloud this has more changes per review comments, and merge conflicts are now resolved. If it looks ok to you guys, please merge for me. Thanks!

**Review by mdlinville — APPROVED** [review]
LGTM

File: docs/reference/glossary.md [hunk 1]
```diff
@@ -62,6 +62,28 @@ be done to get it running.
...
+
+Docker uses a
+[copy-on-write](https://docs.docker.com/engine/userguide/storagedriver/imagesandcontainers/#/the-copy-on-write-strategy)
+technique and a [union file system](#union-file-system) for both images and
+containers to optimize resources and speed performance. Multiple copies of an
+entity share the same instance and each one makes only specific changes to its
+unique layer.
+
+Multiple containers can share access to the same image, and make
+container-specific changes on a writable in-memory layer which is deleted when
```
Comment (thaJeztah) [inline docs/reference/glossary.md]: remove "in-memory" here, because the writable layer is also written to disk

File: docs/reference/glossary.md [hunk 2]
```diff
@@ -100,19 +122,20 @@ lightweight Docker development environment designed specifically for the Mac. A
 native Mac application, Docker for Mac uses the macOS Hypervisor framework,
 networking, and filesystem. It's the best solution if you want to build, debug,
 test, package, and ship Dockerized applications on a Mac. Docker for Mac
-supersedes [Docker Toolbox](#toolbox) as state-of-the-art Docker for macOS.
```
Comment (thaJeztah) [inline docs/reference/glossary.md]: Incomplete sentence; Looks like something went wrong here with the rebase, or the trailing `Docker for Mac` must be removed

File: docs/reference/glossary.md [hunk 3]
```diff
@@ -296,13 +319,25 @@ For Windows 10 systems that support Microsoft Hyper-V (Professional, Enterprise
...
+containers, making them very lightweight and fast.
+
+For more on Docker and union file systems, see [Docker and AUFS in
+practice](https://docs.docker.com/engine/userguide/storagedriver/aufs-driver/),
+[Docker and Btrfs in
+practice](https://docs.docker.com/engine/userguide/storagedriver/btrfs-driver/),
+and [Docker and OverlayFS in
+practice](https://docs.docker.com/engine/userguide/storagedriver/overlayfs-driver/)
+
+Examples implementations of union file systems are
```
Comment (thaJeztah) [inline docs/reference/glossary.md]: s/Examples/Example/

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks @londoncalling! found some minor issues, but LGTM otherwise

**Comment by londoncalling** [comment]
@thaJeztah @mstanleyjones thanks for the reviews! 

Please have a look, and merge if you think all comments are addressed properly and it looks ok.

**Comment by thaJeztah** [comment]
@londoncalling changes look good now, can you squash your commits?

**Comment by londoncalling** [comment]
@thaJeztah how's that?

**Review by thaJeztah — APPROVED** [review]
yup, all good 👍 LGTM

Ho! Ho! here we go

**Pull request merged**